### PR TITLE
191457 Don't restart task when opening push

### PR DESCRIPTION
**Pull request by vvoicehovics**
### Description of Changes

Update launch intent flags for push notifications.

New task + reset if needed are default launcher flags  and mean that:

1. if activity doesn't exist, it's created
2. if it exists in any task anywhere in stack, task is brought to fg and activity put on top

### Breaking Changes

-   None

### Release Checklist

Bump versions in:

-   [x] package.json
-   [x] example/package.json
-   [x] android/src/main/java/com/optimove/reactnative/OptimoveReactNativeInitializer.java
-   [x] ios/OptimoveInitializer.swift
-   [x] CHANGELOG.md

Release:

-   [ ] Squash and merge to main
-   [ ] Delete branch once merged
-   [ ] ~Create tag from main matching chosen version~
-   [ ] Fill out release notes
-   [ ] ~Run `npm publish --access public`~

Holding for 2.0.0 release with other changes.

### CODE REVIEW

File: android/src/main/java/com/optimove/reactnative/PushReceiver.java [hunk 1]
```diff
@@ -31,6 +31,21 @@ protected void onPushOpened(Context context, PushMessage pushMessage) {
...
+  @Override
+  protected Intent getPushOpenActivityIntent(Context context, PushMessage pushMessage) {
+    Intent launchIntent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
+
+    if (null == launchIntent) {
+      return null;
+    }
+
+    launchIntent.putExtra(PushMessage.EXTRAS_KEY, pushMessage);
+    launchIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED);
```
Comment (k-antipochkin) [inline android/src/main/java/com/optimove/reactnative/PushReceiver.java:43]: I think we can remove this and leave the launch method to be decided in the AndroidManifest.  For example, it is defined as singleTop in the example app

Comment (vvoicehovics) [reply android/src/main/java/com/optimove/reactnative/PushReceiver.java:43]: leaving it to consumers gives more flexibility, but also means it has to be set correctly. We probably dont want a default (creates new activity) or singleTop (creates new if launch activity is not on top), but rather smth like "singleTask". having it as flags gives you sensible, consistent behaviour without doing any further work. Aso potentially more fine-grained control, like activity reparenting. not 100% which is better

Comment (k-antipochkin) [reply android/src/main/java/com/optimove/reactnative/PushReceiver.java:43]: Afaik, RN creates a single activity in Android and its launch mode is singleTask, which is ok for this bug. I guess by consistent you mean that we have full control over it, which raises a question - if the user changed it for whatever reason, why overriding it? In any case, those lines cover our use case so we can keep them

**Review by cgwyllie — APPROVED** [review]

File: android/src/main/java/com/optimove/reactnative/PushReceiver.java (hunk 1 above)
Comment (vvoicehovics) [reply android/src/main/java/com/optimove/reactnative/PushReceiver.java]: i didnt know RN adds singleTask by default, then y, these flags shall be ~same. Ideally would let consumers to override everything 

k, let's remove flags and rely on launch mode

**Pull request merged**